fv3core/stencils/dyn_core.py

```diff
@@ -132,29 +136,36 @@ def p_grad_c_stencil(
             )
 
 
-def get_n_con():
-    if spec.namelist.convert_ke or spec.namelist.vtdm4 > 1.0e-4:
-        n_con = spec.grid.npz
+def get_nk_heat_dissipation(namelist, grid):
+    if namelist.convert_ke or namelist.vtdm4 > 1.0e-4:
+        nk_heat_dissipation = grid.npz
     else:
-        if spec.namelist.d2_bg_k1 < 1.0e-3:
-            n_con = 0
+        if namelist.d2_bg_k1 < 1.0e-3:
+            nk_heat_dissipation = 0
         else:
-            if spec.namelist.d2_bg_k2 < 1.0e-3:
-                n_con = 1
+            if namelist.d2_bg_k2 < 1.0e-3:
+                nk_heat_dissipation = 1
             else:
-                n_con = 2
-    return n_con
+                nk_heat_dissipation = 2
+    return nk_heat_dissipation
 
 
-def dyncore_temporaries(shape):
-    grid = spec.grid
+def dyncore_temporaries(shape, namelist, grid):
     tmps = {}
     utils.storage_dict(
         tmps,
         ["ut", "vt", "gz", "zh", "pem", "pkc", "pk3", "heat_source", "divgd"],
         shape,
         grid.full_origin(),
     )
+    if not namelist.hydrostatic:
+        # To write in parallel region, these need to be 3D first
```

Comment:
This comment doesn't make sense to me, could you please either expand it or delete it? What is a "parallel region" and why does it mean these have to be 3D?

Reply:
updated comment